pkg/validation/app.go

```diff
@@ -41,17 +41,17 @@ func (v *Validator) ValidateApp(ctx context.Context, app v1alpha1.App) (bool, er
 		return false, microerror.Mask(err)
 	}
 
-	err = v.validateConfig(ctx, app)
+	err = v.validateLabels(ctx, app)
```

Comment:
Changed the order here so that we discover missing `giantswarm.io/cluster` label early on, before we move on to validating configs that in this scenario relays on the cluster label.